components/odh-notebook-controller/controllers/notebook_runtime.go

```diff
@@ -183,12 +187,21 @@ func extractDisplayName(metadata string) string {
 	return displayName
 }
 
-var multiDash = regexp.MustCompile(`-+`)
+var (
+	invalidChars = regexp.MustCompile(`[^-._a-zA-Z0-9]+`)
+	multiDash    = regexp.MustCompile(`-+`)
+)
 
+// formatKeyName sanitizes display names for use as ConfigMap keys.
+// Returns empty string if input contains only invalid characters.
 func formatKeyName(displayName string) string {
-	s := strings.NewReplacer(" ", "-", "(", "", ")", "", "|", "").Replace(displayName)
-	s = multiDash.ReplaceAllString(strings.ToLower(s), "-")
-	return strings.Trim(s, "-") + ".json"
+	s := invalidChars.ReplaceAllString(strings.ToLower(displayName), "-")
+	s = multiDash.ReplaceAllString(s, "-")
+	s = strings.Trim(s, "-")
+	if s == "" {
+		return ""
+	}
+	return s + ".json"
```

Comment:
would move this up, and add .json in

```
data[formattedName] = metadataParsed
```

would change

```
data[formattedName + ".json"] = metadataParsed
```

Reply:
Yeah, that is also possibility. Technically, we could rename the method to something like "sanitizeName" so it's generic enough and then use it on multiple places eventually.